### DISCUSSION TITLE
How do I use custom navigation for certain routes?

### DISCUSSION THREAD

**Discussion by nandorojo** [vercel/next.js#17234] [Help]
Hey there, quick question. For all routes in `/account`, I would like to use `react-navigation` instead of next router. Is it possible to configure this?

(For context, `react-navigation` functions similarly to `react-router`.)

For example, I would initialize `react-navigation` in the `pages/account.tsx` file, and the component in that file would return a custom router. Then, if you visit `/account`, or `/account/payments`, etc, it’s all is handled by `react-navigation`.

```tsx
// pages/account.tsx

const { Navigator, Screen } = createStackNavigator()

export default () => {
  return (
    <Navigator>
      <Screen name="general" component={General} />
      <Screen name="payments" component={Payments} />
    </Navigator>
  )
}
```

I believe that this would require any requests sent to `/account/*` to be pointed to `/account` for react navigation to handle, correct?

**Comment by weyert** [comment]
Out of interest, are you using Next.js in a React Native app

**Reply by nandorojo** [reply]
Yeah, I’m using React Native Web with next.js and code sharing with my native app.

**Reply by Jawad-LiftOff** [reply]
Hi @nandorojo could you get his working? If yes, could you please share the sample code?

**Reply by nandorojo** [reply]
I did, but unfortunately there is no linking on the pages that use React navigation yet. Not sure if that’s possible or how it would work.

**Reply by nandorojo** [reply]
My solution consists of using the stacks and modals from React navigation within pages only.
